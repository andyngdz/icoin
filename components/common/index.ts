export { ICoinLogo } from 'components/common/Logo'
export { Route } from 'components/common/Route'
