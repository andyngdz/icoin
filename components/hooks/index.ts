export * from './useChartData'
export * from './useGlobal'
export * from './useLivePrice'
export * from './useLivePrices'
export * from './usePagination'
export * from './useTheme'
export * from './useTime'
