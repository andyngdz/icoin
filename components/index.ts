export * from './asset-detail/AssetDetail'
export * from './asset-detail/AssetDetailContent'
export * from './asset-detail/AssetHighLow'
export * from './asset-detail/AssetSummary'
export * from './assets/AssetItem'
export * from './assets/AssetNameAndDescription'
export * from './assets/Assets'
export * from './assets/AssetsTable'
export * from './common/container-wrapper/ContainerWrapper'
export * from './common/logo/Logo'
export * from './common/pagination/Pagination'
export * from './common/paper-wrapper/PaperWrapper'
export * from './common/route/Route'
export * from './common/value-color/ValueColor'
export * from './disclaimer/Disclaimer'
export * from './exchanges/ExchangeContent'
export * from './exchanges/ExchangeItem'
export * from './exchanges/ExchangesAll'
export * from './exchanges/ExchangesAsset'
export * from './exchanges/ExchangeStatus'
export * from './header/Header'
export * from './header/Menu'
export * from './hooks'
export * from './line-chart/LineChart'
export * from './live-prices/LivePrices'
export * from './search/Search'
export * from './symbol-icon/SymbolIcon'
export * from './time-selection/TimeSelection'
export * from './top-bar/TopBar'
export * from './top-bar/TopBarContent'
