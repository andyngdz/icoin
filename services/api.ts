import { Format } from 'services'
import { IAPI, IAssetResponse } from 'types'
import { PER_PAGE } from 'data'
import axios from 'axios'

const CoingeckoAPI = axios.create({
  baseURL: process.env.NEXT_PUBLIC_COINGECKO_API
})

const CoinCapAPI = axios.create({
  baseURL: process.env.NEXT_PUBLIC_COINCAP_API
})

const API: IAPI = {
  getGlobals: () => CoingeckoAPI.get('/global'),

  getAssets: async (
    params = {
      limit: PER_PAGE
    }
  ) => {
    const response = await CoinCapAPI.get<IAssetResponse>('/assets', {
      params
    })
    const { data } = response.data

    response.data.data = Format.toNumber(data, [
      'marketCapUsd',
      'priceUsd',
      'volumeUsd24Hr',
      'changePercent24Hr',
      'supply',
      'maxSupply',
      'vwap24Hr'
    ])

    return response
  }
}

export { API }
