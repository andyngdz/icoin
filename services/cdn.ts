const CDN = {
  getIcon: (id: string) =>
    `https://cdn.jsdelivr.net/gh/atomiclabs/cryptocurrency-icons@9ab8d6934b83a4aa8ae5e8711609a70ca0ab1b2b/svg/color/${id}.svg`,

  defaultIcon: `https://cdn.jsdelivr.net/gh/atomiclabs/cryptocurrency-icons@9ab8d6934b83a4aa8ae5e8711609a70ca0ab1b2b/svg/color/generic.svg`,
}

export { CDN }
