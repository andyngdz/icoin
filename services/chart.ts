/**
 * Author: Andy D. Ng <itc.anhduy@gmail.com>
 * Chart.js contains the useful function to manage a Chart
 */

import { ChartData, ChartOptions, TimeScale } from 'chart.js'
import { IAssetHistory, ICalculateInterval, TTime } from 'types'
import { merge } from 'lodash'
import { sub } from 'date-fns'
import { TIME_TO_INTERVAL } from 'data'
import ChartJS from 'chart.js'

/**
 * @description
 * We are using Roboto font and
 * Set it here for the ChartJS as well
 */
ChartJS.defaults.global.defaultFontFamily = "'Inter', sans-serif"

const Chart = {
  /**
   * @description Calculate interval post data by id, time, start and end date
   *
   * @param id The id of coin such as: bitcoin, ethereum
   *
   * @param time The time from the TimeSelection component - (Check TTime in chart.d.ts)
   *
   * @returns `ICalculateInterval`
   *
   * @example
   * ```
   *    const { time, onTimeChange } = useTime()
   *     const [getIntervals, { data }] = useLazyQuery<IAssetHistories>(COIN_CHART)
   *
   *     useEffect(() => {
   *       const interval = Chart.calculateInterval(id, time)
   *       getIntervals({ variables: interval })
   *     }, [time])
   *```
   */
  calculateInterval: (id: string, time: TTime): ICalculateInterval => {
    let subtractDate: Date
    const end = Date.now()

    switch (time) {
      case '1D':
        subtractDate = sub(end, { days: 1 })
        break
      case '1W':
        subtractDate = sub(end, { weeks: 1 })
        break
      case '1M':
        subtractDate = sub(end, { months: 1 })
        break
      case '3M':
        subtractDate = sub(end, { months: 3 })
        break
      case '6M':
        subtractDate = sub(end, { months: 6 })
        break
      case '1Y':
      case 'ALL':
        subtractDate = sub(end, { years: 1 })
        break
    }

    return {
      end,
      id,
      interval: TIME_TO_INTERVAL[time],
      start: subtractDate.valueOf()
    }
  },

  /**
   * @description Create time scale with unit for Line Chart
   *
   * @param time The time from the TimeSelection component - (Check TTime in chart.d.ts)
   *
   * @returns `TimeScale`
   */
  createTimeScale: (time: TTime): TimeScale => {
    const timeScale: TimeScale = {}

    switch (time) {
      case '1D':
        timeScale.unit = 'hour'
        break
      case '1W':
        timeScale.unit = 'day'
        break
      case '1M':
      case '3M':
        timeScale.unit = 'week'
        break
      case '6M':
      case '1Y':
      case 'ALL':
        timeScale.unit = 'month'
        break
    }

    return timeScale
  },

  /**
   * @description Create a new chart
   *
   * @param ctx The canvas rendering context
   *
   * @param assetHistories Contains the list history data such as: timestamp, priceUsd, etc...
   *
   * @param options The chart options, it will be merged with the default options
   *
   * @returns `ChartJS`
   */
  createNewChart: (
    canvas: HTMLCanvasElement,
    time: TTime,
    chartData: ChartData,
    options?: ChartOptions
  ): ChartJS => {
    const ctx = canvas.getContext('2d')

    return new ChartJS(ctx, {
      type: 'line',
      data: chartData,
      options: merge(
        {
          tooltips: {
            position: 'nearest'
          },
          legend: {
            display: false
          },
          scales: {
            xAxes: [
              {
                type: 'time',
                time: Chart.createTimeScale(time),
                gridLines: {
                  display: false
                }
              }
            ],
            yAxes: [
              {
                position: 'right',
                gridLines: {
                  drawBorder: false
                }
              }
            ]
          }
        },
        options
      )
    })
  },

  /**
   * @description Create chart data
   *
   * @param assetHistories Contains the list history data such as: timestamp, priceUsd, etc...
   *
   * @returns `ChartData`
   */
  createChartData: (assetHistories: Array<IAssetHistory>): ChartData => {
    const data = assetHistories.map(({ timestamp, priceUsd }) => ({
      t: timestamp,
      y: priceUsd
    }))

    const chartData: ChartData = {
      datasets: [
        {
          data
        }
      ]
    }

    return chartData
  }
}

export { Chart }
