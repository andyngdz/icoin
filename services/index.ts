export * from './paginate'
export * from './chart'
export * from './format'
export * from './api'
export * from './cdn'
export * from './test-utils'
export * from './price-direction'
export * from './price'
export * from './routes'
