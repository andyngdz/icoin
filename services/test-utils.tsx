import { Provider } from 'react-redux'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/core'
import React from 'react'
import Store from 'states'
import Theme from 'styles'

const Providers = ({ children }) => {
  return (
    <Provider store={Store}>
      <ThemeProvider theme={Theme}>{children}</ThemeProvider>
    </Provider>
  )
}

const customRender = (ui, options = {}) =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
