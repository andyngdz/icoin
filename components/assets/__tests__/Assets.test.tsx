import { act, customRender, RenderResult } from 'services/test-utils'
import { AssetsContent } from 'components'

describe('Assets', () => {
  it('Should render assets content correctly', async () => {
    let renderResult: RenderResult

    await act(async () => {
      renderResult = customRender(<AssetsContent assets={1600} />)
    })

    expect(renderResult).toMatchSnapshot()
  })
})
