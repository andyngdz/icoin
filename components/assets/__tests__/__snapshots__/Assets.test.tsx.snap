// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assets Should render it correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root ContainerWrapper-container-1 MuiContainer-maxWidthLg"
      >
        <div
          class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation3 MuiPaper-rounded"
        >
          <table
            class="MuiTable-root"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  #
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Name and Description
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                  scope="col"
                >
                  Market Cap
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                  scope="col"
                >
                  Price
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                  scope="col"
                >
                  Volume
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                  scope="col"
                >
                  Change (24h)
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                  scope="col"
                >
                  Circularing Supply
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root AssetItem-asset-2"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2"
                  >
                    1
                    .
                  </h6>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  <div
                    class="AssetNameAndDescription-name-5"
                  >
                    <div>
                      <p
                        class="MuiTypography-root MuiTypography-button"
                      >
                        Bitcoin
                      </p>
                      <span
                        class="MuiTypography-root MuiTypography-caption"
                      >
                        BTC
                      </span>
                    </div>
                  </div>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2"
                  >
                    $647,158,791,754
                  </h6>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2"
                  >
                    $34,761.62
                  </h6>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2"
                  >
                    $14,567,120,537
                  </h6>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  <span
                    class="MuiTypography-root ValueColor-up-6 MuiTypography-subtitle2"
                  >
                    3.15%
                  </span>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2"
                  >
                    18,617,050
                  </h6>
                </td>
              </tr>
              <tr
                class="MuiTableRow-root AssetItem-asset-2"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2"
                  >
                    2
                    .
                  </h6>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                >
                  <div
                    class="AssetNameAndDescription-name-5"
                  >
                    <div>
                      <p
                        class="MuiTypography-root MuiTypography-button"
                      >
                        Ethereum
                      </p>
                      <span
                        class="MuiTypography-root MuiTypography-caption"
                      >
                        ETH
                      </span>
                    </div>
                  </div>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2"
                  >
                    $163,024,735,364
                  </h6>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2"
                  >
                    $1,423.75
                  </h6>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2"
                  >
                    $10,514,268,643
                  </h6>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  <span
                    class="MuiTypography-root ValueColor-up-6 MuiTypography-subtitle2"
                  >
                    7.16%
                  </span>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2"
                  >
                    114,503,686.687
                  </h6>
                </td>
              </tr>
            </tbody>
          </table>
          <div
            class="MuiToolbar-root MuiToolbar-regular Pagination-pagiation-8 MuiToolbar-gutters"
          >
            <nav
              aria-label="pagination navigation"
              class="MuiPagination-root"
            >
              <ul
                class="MuiPagination-ul"
              >
                <li>
                  <button
                    aria-label="Go to previous page"
                    class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiPaginationItem-icon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                      />
                    </svg>
                  </button>
                </li>
                <li>
                  <button
                    aria-current="true"
                    aria-label="page 1"
                    class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-selected"
                    tabindex="0"
                    type="button"
                  >
                    1
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </li>
                <li>
                  <button
                    aria-label="Go to page 2"
                    class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
                    tabindex="0"
                    type="button"
                  >
                    2
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </li>
                <li>
                  <button
                    aria-label="Go to page 3"
                    class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
                    tabindex="0"
                    type="button"
                  >
                    3
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </li>
                <li>
                  <button
                    aria-label="Go to page 4"
                    class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
                    tabindex="0"
                    type="button"
                  >
                    4
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </li>
                <li>
                  <button
                    aria-label="Go to page 5"
                    class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
                    tabindex="0"
                    type="button"
                  >
                    5
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </li>
                <li>
                  <div
                    class="MuiPaginationItem-root MuiPaginationItem-ellipsis"
                  >
                    …
                  </div>
                </li>
                <li>
                  <button
                    aria-label="Go to page 100"
                    class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
                    tabindex="0"
                    type="button"
                  >
                    100
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </li>
                <li>
                  <button
                    aria-label="Go to next page"
                    class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiPaginationItem-icon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root ContainerWrapper-container-1 MuiContainer-maxWidthLg"
    >
      <div
        class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation3 MuiPaper-rounded"
      >
        <table
          class="MuiTable-root"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                #
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Name and Description
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                Market Cap
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                Price
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                Volume
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                Change (24h)
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                Circularing Supply
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root AssetItem-asset-2"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2"
                >
                  1
                  .
                </h6>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                <div
                  class="AssetNameAndDescription-name-5"
                >
                  <div>
                    <p
                      class="MuiTypography-root MuiTypography-button"
                    >
                      Bitcoin
                    </p>
                    <span
                      class="MuiTypography-root MuiTypography-caption"
                    >
                      BTC
                    </span>
                  </div>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2"
                >
                  $647,158,791,754
                </h6>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2"
                >
                  $34,761.62
                </h6>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2"
                >
                  $14,567,120,537
                </h6>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                <span
                  class="MuiTypography-root ValueColor-up-6 MuiTypography-subtitle2"
                >
                  3.15%
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2"
                >
                  18,617,050
                </h6>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root AssetItem-asset-2"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2"
                >
                  2
                  .
                </h6>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                <div
                  class="AssetNameAndDescription-name-5"
                >
                  <div>
                    <p
                      class="MuiTypography-root MuiTypography-button"
                    >
                      Ethereum
                    </p>
                    <span
                      class="MuiTypography-root MuiTypography-caption"
                    >
                      ETH
                    </span>
                  </div>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2"
                >
                  $163,024,735,364
                </h6>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2"
                >
                  $1,423.75
                </h6>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2"
                >
                  $10,514,268,643
                </h6>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                <span
                  class="MuiTypography-root ValueColor-up-6 MuiTypography-subtitle2"
                >
                  7.16%
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2"
                >
                  114,503,686.687
                </h6>
              </td>
            </tr>
          </tbody>
        </table>
        <div
          class="MuiToolbar-root MuiToolbar-regular Pagination-pagiation-8 MuiToolbar-gutters"
        >
          <nav
            aria-label="pagination navigation"
            class="MuiPagination-root"
          >
            <ul
              class="MuiPagination-ul"
            >
              <li>
                <button
                  aria-label="Go to previous page"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiPaginationItem-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                    />
                  </svg>
                </button>
              </li>
              <li>
                <button
                  aria-current="true"
                  aria-label="page 1"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-selected"
                  tabindex="0"
                  type="button"
                >
                  1
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </li>
              <li>
                <button
                  aria-label="Go to page 2"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
                  tabindex="0"
                  type="button"
                >
                  2
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </li>
              <li>
                <button
                  aria-label="Go to page 3"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
                  tabindex="0"
                  type="button"
                >
                  3
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </li>
              <li>
                <button
                  aria-label="Go to page 4"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
                  tabindex="0"
                  type="button"
                >
                  4
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </li>
              <li>
                <button
                  aria-label="Go to page 5"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
                  tabindex="0"
                  type="button"
                >
                  5
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </li>
              <li>
                <div
                  class="MuiPaginationItem-root MuiPaginationItem-ellipsis"
                >
                  …
                </div>
              </li>
              <li>
                <button
                  aria-label="Go to page 100"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
                  tabindex="0"
                  type="button"
                >
                  100
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </li>
              <li>
                <button
                  aria-label="Go to next page"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiPaginationItem-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
